infra/blueprint-test/pkg/krmt/krm.go

```diff
@@ -0,0 +1,312 @@
+		t:          t,
+		setters:    make(map[string]string),
+		updatePkgs: true,
+		timeout:    "10m",
+	}
+	// default KRM blueprint methods
+	krmt.init = krmt.DefaultInit
+	krmt.apply = krmt.DefaultApply
+	krmt.verify = krmt.DefaultVerify
+	krmt.teardown = krmt.DefaultTeardown
+	// apply options
+	for _, opt := range opts {
+		opt(krmt)
+	}
+	// if no custom logger, set default based on test verbosity
+	if krmt.logger == nil {
+		krmt.logger = utils.GetLoggerFromT()
+	}
+	// if explicit bpDir is provided, validate it else try auto discovery
+	if krmt.bpDir != "" {
+		_, err := os.Stat(krmt.bpDir)
+		if os.IsNotExist(err) {
+			t.Fatalf("Dir path %s does not exist", krmt.bpDir)
+		}
+	} else {
+		cwd, err := os.Getwd()
+		if err != nil {
+			t.Fatalf("unable to get wd :%v", err)
+		}
+		bpDir, err := discovery.GetConfigDirFromTestDir(cwd)
+		if err != nil {
+			t.Fatalf("unable to detect KRM dir :%v", err)
+		}
+		krmt.bpDir = bpDir
+	}
+	// if no explicit build directory is provided, setup build directory
+	if krmt.buildDir == "" {
+		krmt.buildDir = krmt.getDefaultBuildDir()
+	}
+	// configure kpt to run in buildDir
+	krmt.kpt = kpt.NewCmdConfig(t, kpt.WithDir(krmt.buildDir))
+	// get well known setters from env vars
+	krmt.getKnownSettersFromEnv()
+
+	krmt.logger.Logf(krmt.t, "Running tests KRM configs in %s", krmt.bpDir)
+	return krmt
+}
+
+// getDefaultBuildDir returns a temporary build directory for hydrated configs.
+func (b *KRMBlueprintTest) getDefaultBuildDir() string {
+	cwd, err := os.Getwd()
+	if err != nil {
+		b.t.Fatalf("unable to get wd :%v", err)
+	}
+	buildDir := path.Join(cwd, tmpBuildDir)
+	err = os.MkdirAll(buildDir, 0755)
+	if err != nil {
+		b.t.Fatalf("unable to create %s :%v", buildDir, err)
+	}
+	exampleBuildDir := path.Join(buildDir, b.t.Name())
+	abs, err := filepath.Abs(exampleBuildDir)
+	if err != nil {
+		b.t.Fatalf("unable to get absolute path for %s :%v", exampleBuildDir, err)
+	}
+	return abs
+}
+
+// getKnownSettersFromEnv creates setters from known CommonSetters env vars.
+func (b *KRMBlueprintTest) getKnownSettersFromEnv() {
+	setters := make(map[string]string)
+	for _, s := range CommonSetters {
+		sKey, sVal, err := kpt.GenerateSetterKVFromEnvVar(s)
+		// if a common env var is not set, log and continue
+		if err != nil {
+			b.logger.Logf(b.t, "Skipping env var %s: %v", s, err)
+		} else {
+			b.logger.Logf(b.t, "Setting env var %s as setter %s: %s", s, sKey, sVal)
+			setters[sKey] = sVal
+		}
+	}
+	// merge user provided setters with env discovered setters
+	// user provided setters override env discovered setters
+	b.setters = kpt.MergeSetters(setters, b.setters)
+}
+
+// ValFromEnv returns value for a given env var, fails test if env var is not set.
+func (b *KRMBlueprintTest) ValFromEnv(k string) string {
+	v, err := utils.ValFromEnv(k)
+	if err != nil {
+		b.t.Fatal(err)
+	}
+	return v
+}
+
+// setupBuildDir prepares build dir with configs from bpDir.
```

Comment:
Isn't this actually configs from example/fixture directory (not the blueprint itself)?

Reply:
Yeah its the configs in the example dir. This name was misleading so I switched it to `exampleDir`.